pkg/controller/nodeipam/ipam/multinetwork_cloud_cidr_allocator.go

```diff
@@ -42,6 +61,11 @@ func (ca *cloudCIDRAllocator) performMultiNetworkCIDRAllocation(node *v1.Node, i
 			rangeNameAliasIPMap[ipRange.SubnetworkRangeName] = ipRange
 		}
 		for _, network := range networks {
+			if _, ok := matchedNetworks[network.Name]; ok {
+				// skip networks that are already matched with an interface
+				continue
+			}
+
```

Comment:
I didn't get when we would hit this condition and continue.

Reply:
since outer loop is at interface level, (inf0 -- {nw1,nw2,nw3}) and let's say inf0 matched with nw3, we don't need to check for nw3 when we process next interface (inf1)